go-controller/pkg/ovn/controller/services/services_controller.go

```diff
@@ -544,16 +544,38 @@ func (c *Controller) RequestFullSync(nodeInfos []nodeInfo) {
 
 // handlers
 
+// skipService is used when UDN is enabled to know which services are to be skipped because they don't
+// belong to the network that this service controller is responsible for.
+func (c *Controller) skipService(name, namespace string) bool {
+	if util.IsNetworkSegmentationSupportEnabled() {
+		serviceNetwork, err := util.GetActiveNetworkForNamespace(namespace, c.nadLister)
+		if err != nil {
+			utilruntime.HandleError(fmt.Errorf("failed to retrieve network for service %s/%s: %w",
+				namespace, name, err))
+			return true
+		}
+		if serviceNetwork.GetNetworkName() != c.netInfo.GetNetworkName() {
+			return true
+		}
+	}
+
+	return false
+}
+
 // onServiceAdd queues the Service for processing.
 func (c *Controller) onServiceAdd(obj interface{}) {
 	key, err := cache.MetaNamespaceKeyFunc(obj)
 	if err != nil {
-		utilruntime.HandleError(fmt.Errorf("couldn't get key for object %+v: %v", obj, err))
+		utilruntime.HandleError(fmt.Errorf("couldn't get key for object %+v for network=%s: %v", obj, c.netInfo.GetNetworkName(), err))
 		return
 	}
-	klog.V(5).Infof("Adding service %s", key)
+
 	service := obj.(*v1.Service)
+	if c.skipService(service.Name, service.Namespace) {
```

Comment:
So this will also skip for the default services controller for any services that are on UDN  networks right?

Reply:
yes :)